src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsuofdataapi/integration/service/ReportServiceCoroutinesTest.kt

```diff
@@ -89,4 +91,95 @@ class ReportServiceCoroutinesTest {
 
     assertEquals(gotHmppsSubjectAccessRequestContent, expectedHmppsSubjectAccessRequestContent)
   }
+
+  @Test
+  fun `test getReportsForSubjectAccess returns correct reports coroutines runTest with only fromDate `() = runTest {
+    val gotListReportDetail = listOf(report1)
+    val listReportDetail = listOf(report1)
+
+    whenever(
+      reportRepository.findAllByOffenderNoAndIncidentDateAfter(
+        offenderNumber,
+        fromDate.atStartOfDay(),
+      ),
+    ).thenReturn(gotListReportDetail)
+
+    val reports = reportService.getReportsForSubjectAccess(offenderNumber, fromDate, null)
+    assertEquals(listReportDetail, gotListReportDetail)
+    assert(reports.size == 1)
+  }
+
+  @Test
+  fun `test getReportsForSubjectAccess returns correct reports coroutines runTest with only toDate `() = runTest {
+    val listReportDetail = listOf(report1, report2)
+
+    val expectedHmppsSubjectAccessRequestContent = HmppsSubjectAccessRequestContent(
+      content = listOf(report1, report2).map {
+        it.toDto(
+          includeStatements = true,
+          includeFormResponse = true,
+        )
+      },
+    )
+
+    whenever(
+      reportRepository.findAllByOffenderNoAndIncidentDateBefore(
+        offenderNumber,
+        toDate.plusDays(1).atStartOfDay(),
+      ),
+    ).thenReturn(listOf(report1, report2))
+
+    whenever(reportServiceMock.getReportsByOffenderNumberAndDateWindow(offenderNumber, fromDate, toDate)).thenReturn(
+      listReportDetail,
+    )
+    val gotHmppsSubjectAccessRequestContent = reportService.getPrisonContentFor(offenderNumber, null, toDate)
+    assertNotNull(listReportDetail)
```

Comment:
is that what you want to test? since you create it above, can it really be null?